photon-api/src/test/scala/com/linkedin/photon/ml/data/LocalDataSetTest.scala

```diff
@@ -51,6 +53,254 @@ class LocalDataSetTest {
+      meanAbs = Vector(0.0),
+      None: Option[Int]
+    )
+    val t_bt_1_selected_g_positive = Array(5.0)
+
+    // Seventh case t < 1 but not selected
+    val t_lt_1_not_selected = Array(
+      SparseVector(1.0),
+      SparseVector(1.0),
+      SparseVector(1.0),
+      SparseVector(1.0),
+      SparseVector(1.0)
+    )
+    val t_lt_1_not_selected_l = Array(0.0, 0.0, 0.0, 0.0, 1.0)
+
+    val t_lt_1_not_selected_expected = Map(0 -> 0.140309013)
+
+    val t_lt_1_not_selected_stats = BasicStatisticalSummary(
+      mean = Vector(1.0),
+      variance = Vector(0.0),
+      count = 50,
+      numNonzeros = Vector(50.0),
+      max = Vector(0.0),
+      min = Vector(0.0),
+      normL1 = Vector(0.0),
+      normL2 = Vector(0.0),
+      meanAbs = Vector(0.0),
+      None: Option[Int]
+    )
+    val t_lt_1_not_selected_g_positive = Array(15.0)
+
+    // Eighth case t < 1 selected
+    val t_lt_1_selected = Array.fill[Vector[Double]](30)(Vector(1.0))
+
+    val t_lt_1_selected_l = Array.fill[Double](30)(0.0)
+    Seq(0,1,2).foreach(t_lt_1_selected_l(_) = 1)
+
+    val t_lt_1_selected_expected = Map(0 -> 1.816122265)
+
+    val t_lt_1_selected_stats = BasicStatisticalSummary(
+      mean = Vector(1.0),
+      variance = Vector(0.0),
+      count = 100,
+      numNonzeros = Vector(100.0),
+      max = Vector(0.0),
+      min = Vector(0.0),
+      normL1 = Vector(0.0),
+      normL2 = Vector(0.0),
+      meanAbs = Vector(0.0),
+      None: Option[Int]
+    )
+    val t_lt_1_selected_g_positive = Array(75.0)
+
+    Array(
+      Array[Any](x_0_y_0, x_0_y_0_l, x_0_y_0_expected, x_0_y_0_stats, binaryIndices , nonBinaryIndices, x_0_y_0_g_positive, 0),
+      Array[Any](x_0_y_not_0, x_0_y_not_0_l, x_0_y_not_0_expected, x_0_y_not_0_stats, binaryIndices, nonBinaryIndices, x_0_y_not_0_g_positive, 0),
+      Array[Any](nonBinary, nonBinary_l, nonBinary_expected, nonBinary_stats, binaryIndicesWithNonBinary, nonBinaryIndicesWithNonBinary, nonBinary_g_positive, 1),
+      Array[Any](x_m_y_n, x_m_y_n_l, x_m_y_n_expected, x_m_y_n_stats, binaryIndices, nonBinaryIndices, x_m_y_n_g_positive, 0),
+      Array[Any](t_bt_1_not_selected, t_bt_1_not_selected_l, t_bt_1_not_selected_expected, t_bt_1_not_selected_stats, binaryIndices, nonBinaryIndices, t_bt_1_not_selected_g_positive, 0),
+      Array[Any](t_bt_1_selected, t_bt_1_selected_l, t_bt_1_selected_expected, t_bt_1_selected_stats, binaryIndices, nonBinaryIndices, t_bt_1_selected_g_positive, 1),
+      Array[Any](t_lt_1_not_selected, t_lt_1_not_selected_l, t_lt_1_not_selected_expected, t_lt_1_not_selected_stats, binaryIndices, nonBinaryIndices, t_lt_1_not_selected_g_positive, 0),
+      Array[Any](t_lt_1_selected, t_lt_1_selected_l, t_lt_1_selected_expected, t_lt_1_selected_stats, binaryIndices, nonBinaryIndices, t_lt_1_selected_g_positive, 1)
+    )
+  }
+
+  @Test(dependsOnGroups = Array[String]("testComputeRatioCILowerBound", "testCore"),
+    dataProvider = "dataForRandomEffectFeatureSelection")
```

Comment:
Currently `testFilterFeaturesByRatioCIBound` and `testComputeRatioCILowerBound` take the same input and repeat the same tests.

Instead, this test should take as input a consistent dataset and global feature data. The data provider should vary the binary indices set, non-binary indices set, filtering threshold, and zScore. The test cases should be something like this:

1. All binary features selected
2. Modify binary features set: one of the binary features is no longer selected
3. Modify non-binary features set: one of the non-binary features is no longer selected
4. Modify the filtering threshold: one of the binary features is no longer selected
5. Modify the z-score: one of the binary features is no longer selected